core/src/env_platform/db_handler.bal

```diff
@@ -155,7 +156,68 @@ function updateApplicationDraft(TreeRemovalForm form, string applicationId) retu
         log:printDebug("Updated status for application with application ID: " + applicationId + " is " + updated.toString() + ".");
         return updated == 1 ? true : false;
     } else {
-        log:printDebug("An error occurred while updating the draft application with the application ID: " + applicationId + ". " + updated.reason() + ".");
+        log:printDebug("An error occurred while updating the draft application with the application ID: " + applicationId + ". " + updated.reason().toString() + ".");
+        return updated;
+    }
+}
+
+# The `updateApplication` function will add a new version of an application with the incoming form details.
+# 
+# + form - Form containing the tree removal data.
+# + applicationId - The Id of the application which should be added.
+# + return - This function will return true if version of the application is updated in the database, 
+# false if not or else it returns a mongodb:Database error.
+function updateApplication(TreeRemovalForm form, string applicationId) returns boolean|error {
+
+    // Get the exsisting application.
+    map<json>[] found = check applicationCollection->find({"applicationId": applicationId});
+    log:printDebug("The application of application id: " + applicationId.toString() + " is " + found.toString());
+
+    // Get the versions array.
+    json[] versions = <json[]>found[0].versions;
+
+    map<json> application = {
+        "title": form.title,
+        "applicationCreatedDate": {
+            "year": form.applicationCreatedDate.year,
+            "month": form.applicationCreatedDate.month,
+            "day": form.applicationCreatedDate.day,
+            "hour": form.applicationCreatedDate.hour,
+            "minute": form.applicationCreatedDate.minute
+        },
+        "removalDate": {
+            "year": form.removalDate.year,
+            "month": form.removalDate.month,
+            "day": form.removalDate.day,
+            "hour": form.removalDate.hour,
+            "minute": form.removalDate.minute
+        },
+        "reason": form.reason,
+        "applicationType": form.applicationType,
+        "requestedBy": form.requestedBy,
+        "permitRequired": form.permitRequired,
+        "landOwner": form.landOwner,
+        "treeRemovalAuthority": form.treeRemovalAuthority,
+        "city": form.city,
+        "district": form.district,
+        "nameOfTheLand": form.nameOfTheLand,
+        "planNumber": form.planNumber,
+        "area": extractAreaAsJSONArray(form.area),
+        "treeInformation": extractTreeInformationAsJSONArray(form.treeInformation)
+    };
+    log:printDebug("Constructed application: " + application.toString());
+
+    // Add the new version to the versions array
+    versions.push(application);
```

Comment:
Since the only real difference between **updateApplicationDraft** method and **updateApplication** is this line and the versions array definition, I suggest removing this newly added **updateApplication** function and moving this line to the updateApplicationDraft method. We could pass in a **boolean value** to figure out whether the update _is for making a new version of the application or not_. 

@yashodgayashan @deshankoswatte @vikumkbv let me know if there is a reason if this change is not possible.

Reply:
Yeah it can be done. Since update application draft will alter the existing version but update application will get the versions array and push the new version to the versions array and update the application document. Shall we make a one function ?

Reply:
+1 for @sanjulamadurapperuma's comment.

Reply:
+1 to @sanjulamadurapperuma's comment